sed/io.py

```diff
@@ -1 +1,163 @@
-# TODO: populate
+import h5py
+import xarray as xr
+import numpy as np
+
+def recursive_write_metadata(h5group, node):
+    for key, item in node.items():
+        if isinstance(item, (np.ndarray, np.int64, np.float64, str, bytes, int, float, list)):
+            try:
+                h5group.create_dataset(key, data=item)
+            except TypeError:
+                h5group.create_dataset(key, data=str(item))
+                print("saved " + key + " as string")
+        elif isinstance(item, dict):
+            print(key)
+            group = h5group.create_group(key)
+            recursive_write_metadata(group, item)
+        else:
+            try:
+                h5group.create_dataset(key, data=str(item))
+                print("saved " + key + " as string")
+            except:
+                raise ValueError('Cannot save %s type'%type(item))
+
+def recursive_parse_metadata(node):
+    if isinstance(node, h5py.Group):
+        dictionary = {}
+        for key, value in node.items():
+            dictionary[key] = recursive_parse_metadata(value)
+
+    else:
+        dictionary = node[...]
+        try:
+            dictionary = dictionary.item()
+            if isinstance(dictionary, (bytes, bytearray)):
+                dictionary = dictionary.decode()
+        except ValueError:
+            pass
+
+    return dictionary
+
+
+DEFAULT_UNITS = {
+    'X': 'step',
+    'Y': 'step',
+    't': 'step',
+    'tofVoltage': 'V',
+    'extractorVoltage': 'V',
+    'extractorCurrent': 'A',
+    'cryoTemperature': 'K',
+    'sampleTemperature': 'K',
+    'dldTimeBinSize': 'ns',
+    'delay': 'ps',
+    'timeStamp': 's',
+    'energy': 'eV',
+    'kx': '1/A',
+    'ky': '1/A'}
+
+def res_to_xarray(res, bin_names, bin_axes, metadata=None):
+    """ creates a BinnedArray (xarray subclass) out of the given np.array
+    Parameters:
```

Comment:
For consistency, IMHO it's better to use a single style for docstrings

Reply:
I'll try to keep the style of the binning.py. Is this somehow auto-formatted?

Reply:
I used the `google-notypes` style (google, which does not add the types in the comments), and auto-formatting exists for visual studio code as the extension autoDocstring, or somewhere in the settings of pycharm.

Reply:
> I'll try to keep the style of the binning.py. Is this somehow auto-formatted?

What I meant with that is the following:
- Start the docstring with capitalize letters.
- Use "camel_case" for naming variables, functions, ...
- In the `binning` module, "Args" is used instead of "Parameters"

I understand that these are "nitpicking" issues, but it's better to addressed them now that there is little code written :)

Reply:
Fair points, I will go through once more.